docs/cow-protocol/tutorials/cow-swap/flash-loans.mdx

```diff
@@ -0,0 +1,264 @@
+- [Token ratios](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#4-determine-token-amounts-needed-to-trade-and-deposit) may need to be balanced, and trading amounts adjusted accordingly.
+- Tokens in Aave can be [obtained from the faucet](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#5-get-tokens-from-the-faucet).
+- Check [withdrawal and borrowing availability](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#7-check-aaves-withdrawalborrow-availability) of the tokens on Aave.
+
+In a production network (e.g. mainnet), these considerations generally won’t apply.
+
+:::
+
+## Aave repay pre-hook
+We need to define a [pre-hook](/cow-protocol/reference/core/intents/hooks) to repay the 5,000 USDT debt on Aave.
+
+As part of the settlement, we want to call the [`repay` function on Aave's pool](https://aave.com/docs/developers/smart-contracts/pool#write-methods-repay):
+```solidity
+function repay(
+    address asset,
+    uint256 amount,
+    uint256 interestRateMode,
+    address onBehalfOf
+) public virtual override returns (uint256)
+```
+
+The following Typescript code generates the corresponding function data using [`viem`](https://viem.sh/):
+```typescript
+const repayTxData = encodeFunctionData({
+    abi, // ABI definition for the repay function
+    functionName: 'repay',
+    args: [
+      '0xaA8E23Fb1079EA71e0a56F48a2aA51851D8433D0', // asset: USDT
+      '5000000000', // amount: 5,000 USDT (USDT has 6 decimals)
+      2, // interestRateMode: 2 for variable
+      '0x35eD9A9D1122A1544e031Cc92fCC7eA599e28D9C', // onBehalfOf: Safe address
+    ]
+  });
+```
+
+The `repay` call must be executed during the order settlement process. To enable this, the pre-hook defines a transaction that will be sent to the Safe wallet. The Safe then executes the actual repay call to the Aave pool.
+The transaction can be built using the [Safe Protocol Kit](https://docs.safe.global/sdk/protocol-kit) as shown below:
+```typescript
+  const repaySafeTxData: SafeTransactionDataPartial = {
+    to: '0x6Ae43d3271ff6888e7Fc43Fd7321a503ff738951', // Aave pool address in Sepolia
+    value: '0',
+    data: repayTxData, // created previously
+    operation: OperationType.Call,
+    nonce, // current Safe nonce + 1
+  };
+
+  const safeTransaction = await safe.createTransaction({ transactions: [repaySafeTxData] });
+  const signedSafeTransaction = await safe.signTransaction(safeTransaction, SigningMethod.ETH_SIGN);
+  const encodedSafeTransaction = await safe.getEncodedTransaction(signedSafeTransaction);
+```
+
+:::note
+For this example, the `nonce` should be offset by 1 from the current Safe's nonce, as this transaction will be executed in the future.
+This is needed to avoid using the same nonce twice, as we will send another transaction for the pre-signature in following sections.
```

Comment:
Phrasing is slightly weird. Maybe this gives slightly more context.
```suggestion
This is needed to avoid using the same nonce twice, as we will send another transaction to emit the pre-signature for our order which is needed to make it executable in the first place.
```